src/main/java/spring/data/access/service/CartDao.java

```diff
@@ -0,0 +1,26 @@
+package spring.data.access.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import spring.data.access.entity.Cart;
+import spring.data.access.repository.JDBCTemplateCart;
+
+@Component
+@Scope("prototype")
+public class CartDao {
+    @Autowired
```

Comment:
you need to have try/catch in all methods

Reply:
methods have been surrounded with try & catch.